pjlib/include/pj/assert.h

```diff
@@ -40,14 +40,50 @@
  * Check during debug build that an expression is true. If the expression
  * computes to false during run-time, then the program will stop at the
  * offending statements.
- * For release build, this macro will not do anything.
+ * For release build, this macro will only log the assertion, while the
+ * program continues running.
  *
  * @param expr      The expression to be evaluated.
  */
-#ifndef pj_assert
-#   define pj_assert(expr)   assert(expr)
+#if PJ_DEBUG==0
+
+#   ifndef pj_assert
+#       include "pj/log.h"
+#       define pj_assert(expr) \
+            do { \
+                if (!(expr)) { \
+                    PJ_LOG(1, (__FILE__, "Assert failed: %s", #expr)); \
+                } \
+            } while (0)
+#   endif
+
+#else /* PJ_DEBUG != 0 */
+
+#   ifndef pj_assert
+#       define pj_assert(expr)   assert(expr)
+#   endif
+
 #endif
 
+/**
+ * @hideinitializer
+ * For all builds, log the message.
+ * Check during debug build that an expression is true. If the expression
+ * computes to false during run-time, then the program will stop at the
+ * offending statements.
+ * For release build, this macro only print message on the log.
+ * @param expr	    The expression to be evaluated.
+ * @param ...	    File name. The format string for the log message
+ *                  ("config.c", " PJ_VERSION: %s", PJ_VERSION)
+ */
+#ifndef PJ_ASSERT_LOG
```

Comment:
This doesn't seem to be used anywhere. Should we remove it?

Reply:
As it looks simple and can be useful for some apps/users, perhaps it is okay?